File: documentation-examples-java/src/java/dev/typr/foundations/docs/mariadb/DomainType.java
package dev.typr.foundations.docs.mariadb;

import dev.typr.foundations.MariaType;
import dev.typr.foundations.MariaTypes;

@SuppressWarnings("unused")
public class DomainType {
  // start
  // Wrapper type
  public record UserId(Long value) {}

  // Create MariaType from bigint
  MariaType<UserId> userIdType = MariaTypes.bigint.transform(UserId::new, UserId::value);
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/mariadb/DomainType.scala
package dev.typr.foundationssc.docs.mariadb
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

@SuppressWarnings(Array("unused"))
object DomainType:
  // start
  // Wrapper type
  case class UserId(value: Long)

  // Create MariaType from bigint
  val userIdType: MariaType[UserId] = MariaTypes.bigint.transform(UserId.apply, _.value)
  // stop
